query-translator/tests/translate.rs

```diff
@@ -466,3 +466,39 @@ fn test_order_by() {
                      ORDER BY `?y_value_type_tag` ASC, `?y` ASC, `?x` ASC");
     assert_eq!(args, vec![]);
 }
+
+#[test]
+fn test_complex_nested_or_join_type_projection() {
+    let mut schema = Schema::default();
+    associate_ident(&mut schema, NamespacedKeyword::new("page", "title"), 98);
+    add_attribute(&mut schema, 98, Attribute {
+        value_type: ValueType::String,
+        ..Default::default()
+    });
+    associate_ident(&mut schema, NamespacedKeyword::new("page", "title2"), 97);
+    add_attribute(&mut schema, 97, Attribute {
+        value_type: ValueType::String,
+        ..Default::default()
+    });
+
+    let input = r#"[:find [?y]
+                    :where
+                    (or
+                      (or
```

Comment:
This test is a little fragile, since it's likely we'll eventually optimize this away:
1) a repeat `or` can be dropped;
2) even if it's not a repeat `or`, we can turn this into a single `WHERE a = 98 OR a = 99`.